azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -454,29 +457,31 @@ public void run() {
       // So we set it to KILLED to make sure we know that we forced kill it
       // rather than
       // it being a legitimate failure.
-      changeStatus(Status.KILLED);
+      finalStatus = changeStatus(Status.KILLED);
     }
 
     int attemptNo = node.getAttempt();
     logInfo("Finishing job " + this.jobId + " attempt: " + attemptNo + " at "
         + node.getEndTime() + " with status " + node.getStatus());
 
-    fireEvent(Event.create(this, Type.JOB_FINISHED), false);
+    // TODO THIS IS THE PLACE FOR SURE
+    fireEvent(Event.create(this, Type.JOB_FINISHED, EventData.create(finalStatus)), false);
     finalizeLogFile(attemptNo);
     finalizeAttachmentFile();
     writeStatus();
   }
 
-  private boolean prepareJob() throws RuntimeException {
+  private Status prepareJob() throws RuntimeException {
     // Check pre conditions
     if (props == null || this.isKilled()) {
       logError("Failing job. The job properties don't exist");
-      return false;
+      return null;
     }
 
+    Status finalStatus;
     synchronized (syncObject) {
       if (node.getStatus() == Status.FAILED || this.isKilled()) {
-        return false;
+        return null;
```

Comment:
Same private method (`prepareJob()`). No need for specific testing in my opinion.